### PR TITLE
cv 0.7.1

### PR DESCRIPTION
**Pull request by guoxiao**
moving from `homebrew/homebrew-headonly`

### CODE REVIEW

File: Library/Formula/cv.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Cv < Formula
+  homepage "https://github.com/BestPig/cv"
+  desc "Coreutils Viewer: Show progress for cp, rm, dd..."
```
Comment (bfontaine) [inline Library/Formula/cv.rb]: Please move this line above the homepage.

**Comment by bfontaine** [comment]
Please make a PR to [Homebrew/homebrew-headonly](https://github.com/Homebrew/homebrew-head-only) to remove the formula there.

File: Library/Formula/cv.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class Cv < Formula
+  desc "Coreutils Viewer: Show progress for cp, rm, dd..."
```
Comment (DomT4) [inline Library/Formula/cv.rb]: Let's not use `...` in this context - It looks like there should be something else printed afterwards in the command-line environment. Perhaps just `Progress meter for coreutils commands` or something?

File: Library/Formula/cv.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+class Cv < Formula
+  desc "Coreutils Viewer: Show progress for coreutils"
+  homepage "https://github.com/BestPig/cv"
```
Comment (nwinkler) [inline Library/Formula/cv.rb]: This is still pointing to the fork... see #32161.

Shouldn't this point to the official repo instead? https://github.com/Xfennec/cv

Comment (guoxiao) [inline Library/Formula/cv.rb]: I've notices it. The official repo just merged BestPig's OSX support patch, but it's not included in latest release. So it still uses BestPig's fork.

Comment (nwinkler) [inline Library/Formula/cv.rb]: True - you're right!

Comment (mikemcquaid) [inline Library/Formula/cv.rb]: We won't include this until we can have it pointing to the non-forked version, sorry!

Comment (guoxiao) [inline Library/Formula/cv.rb]: OK I have requested the author to create a new release supporting OSX.
see Xfennec/cv#45

**Comment by guoxiao** [comment]
I've updated the formula to use official release and BestPig's patch.

File: Library/Formula/cv.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    url "https://patch-diff.githubusercontent.com/raw/Xfennec/cv/pull/19.diff"
+    sha256 "3b3b702ad6d51131ef235ebd3544709d28d7f0d39a0fabe833f4909202819b76"
+  end
+
+  def install
+    system "make", "PREFIX=#{prefix}", "install"
+  end
+
+  test do
+    system "cv", "--help"
```
Comment (mikemcquaid) [inline Library/Formula/cv.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Library/Formula/cv.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
+class Cv < Formula
+  desc "Coreutils Viewer: Show progress for coreutils"
+  homepage "https://github.com/Xfennec/cv"
+  url "https://github.com/Xfennec/cv/archive/v0.6.tar.gz"
+  sha256 "4291ab2f0c507df9eb694bf956ed65dc12bae3c492b795235142b5b87e2e2239"
+  head "https://github.com/Xfennec/cv.git"
+
+  patch do
+    url "https://patch-diff.githubusercontent.com/raw/Xfennec/cv/pull/19.diff"
```
Comment (mikemcquaid) [inline Library/Formula/cv.rb]: Mind making this `https://github.com/Xfennec/cv/pull/19.patch`? It just makes it easier to jump to the actual PR.

File: Library/Formula/cv.rb (hunk 4 above)
Comment (guoxiao) [inline Library/Formula/cv.rb]: It's difficult to write a test, because cv is a monitor of other programmes(cp, dd, etc).

Comment (mikemcquaid) [inline Library/Formula/cv.rb]: Could you just call one of those other programs with e.g. copying a zero-byte file?

Comment (guoxiao) [inline Library/Formula/cv.rb]: The time of copying a zero-byte file may be too short to be monitored..

Comment (mikemcquaid) [inline Library/Formula/cv.rb]: You could `dd` to `/dev/null`, fork, `cv` and then kill the `dd`.

Comment (guoxiao) [inline Library/Formula/cv.rb]: Good idea. I've updated the test.

File: Library/Formula/cv.rb [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  test do
+    system "cv", "--help"
+    pid = fork do
+      system "/bin/dd", "if=/dev/zero", "of=/dev/null", "bs=100000", "count=1000000"
+    end
+    sleep 1
+    begin
+      assert shell_output("cv").include?("dd")
+    ensure
+      Process.kill 9, pid
```
Comment (mikemcquaid) [inline Library/Formula/cv.rb]: Does a regular `kill` not work here?

File: Library/Formula/cv.rb [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+      Process.wait pid
```
Comment (mikemcquaid) [inline Library/Formula/cv.rb]: Might not be worth waiting here?

File: Library/Formula/cv.rb [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  def install
+    system "make", "PREFIX=#{prefix}", "install"
+  end
+
+  test do
+    system "cv", "--help"
+    pid = fork do
+      system "/bin/dd", "if=/dev/zero", "of=/dev/null", "bs=100000", "count=1000000"
+    end
+    sleep 1
```
Comment (mikemcquaid) [inline Library/Formula/cv.rb]: Why is the sleep needed? Not disagreeing, just curious.

File: Library/Formula/cv.rb [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    begin
+      assert shell_output("cv").include?("dd")
```
Comment (mikemcquaid) [inline Library/Formula/cv.rb]: `assert_match /dd/, shell_output("cv")`

File: Library/Formula/cv.rb (hunk 8 above)
Comment (guoxiao) [inline Library/Formula/cv.rb]: Only for fear that `dd` is not ready when `cv` executed. It's not needed and I've removed it.

**Comment by guoxiao** [comment]
The upstream has a new release supporting OS X. 
Updated to release 0.7.1.

File: Library/Formula/cv.rb (hunk 8 above)
Comment (guoxiao) [inline Library/Formula/cv.rb]: It seems that @BrewTestBot failed after `sleep 1` was removed. I re added it.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for new formulae is `cv 0.7.1 (new formula)`.

**Pull request closed without merging**